### DISCUSSION TITLE
Hetero GNN Link Prediction with LinkNeighborLoader

### DISCUSSION THREAD

**Discussion by fenglangsss** [pyg-team/pytorch_geometric#4707] [Q&A] [answered]
I would like to use heterogeneous graph neural networks for the link prediction task. Since my graph is rather large, I want to use LinkNeighborLoader to train the edges in batches. However, I found that LinkNeighborLoader can only load on a single type of edge when instantiated on a heterogeneous graph. What can I do if I want to train on multiple types of edges?

**Comment by rusty1s** [comment, marked as answer]
This is currently not possible as it heavily complicates the underlying computation, e.g., we will now have `edge_label_index` and `edge_label` across multiple edge types, and need to sample negatives accordingly. During training, we need to compute heads for all potential edge types, while some of them may not even be present in the current mini-batch. I definitely see the use-cases for this, but I am not yet sure on how to best integrate this feature. cc @Padarn for any additional thoughts.

**Reply by Padarn** [reply]
I guess a simple workaround would be to create multiple loaders and put the graphs together in a batch, but it's not a very efficient solution.

**Reply by rusty1s** [reply]
I remember long time ago that you wanted to bring support for multiple node types to `NeighborLoader`. Do you think it is worth to come up with a roadmap for this (and potential implications)?

**Reply by Padarn** [reply]
It wasn't that long ago :-) But yeah thats a good idea - I've got a couple of items to close up but I think after that it would be a nice thing to pick up!

**Reply by Padarn** [reply]
I've started a draft of this here https://github.com/pyg-team/pytorch_geometric/issues/4765 - would appreciate your input @fenglangsss!
